### DISCUSSION TITLE
Others get a white screen when they want to view a public accessible folder with CBR

### DISCUSSION THREAD

**Discussion by CrazyGoGo42** [devnoname120/epubviewer#17] [General] [answered]
https://nextcloud.jolineobrien.com/s/eJ8rZFXKQxEs4Dy

Open link without account
Click on CBR file, epubviewer opens in new tab
Does not load the cbr file.

**Comment by devnoname120** [comment]
Need to fix the CBR filename parsing code as it's super fragile.
![image](https://github.com/devnoname120/epubviewer/assets/2824100/f4bdbba7-e51a-45f8-9623-cfe21fb6da8a)

**Comment by roru69** [comment]
Hi,

When i open a CBR file that is "locally" available in Nextcloud, all is good. The file opens in a separate TAB and i can read the comicbook. This is the url it creates: (domain and user replaced by: nextcloud.mydomain.net / username)

"https://nextcloud.mydomain.net/index.php/apps/epubviewer/?file=%2Fremote.php%2Fdav%2Ffiles%2Fusername%2FBlake%2520En%2520Mortimer%2520-%252001%2520-%2520Het%2520Geheim%2520Van%2520De%2520Zwaardvis%2520Deel%25201.cbr&type=application%2Fcomicbook%2Brar"

"https://nextcloud.mydomain.net/index.php/apps/epubviewer/?file=/remote.php/dav/files/username/Blake En Mortimer - 01 - Het Geheim Van 0De Zwaardvis Deel 1.cbr&type=application/comicbook+rar"

But when i try to open the same file that is on a external share (nas01) in Nextcloud, another tab opens, but the screen stays white. The comic doesn't open. This is the URL it creates. (domain and user replaced by: nextcloud.mydomain.net / username)

"https://nextcloud.mydomain.net/index.php/apps/epubviewer/?file=%2Fremote.php%2Fdav%2Ffiles%2Fusername%2F%23EBooks%2520op%2520NAS01%2Fcomics%2FBlake%2520en%2520Mortimer%2FBlake%2520En%2520Mortimer%2520-%252001%2520-%2520Het%2520Geheim%2520Van%2520De%2520Zwaardvis%2520Deel%25201.cbr&type=application%2Fcomicbook%2Brar"

"https://nextcloud.mydomain.net/index.php/apps/epubviewer/?file=/remote.php/dav/files/username/#EBooks op NAS01/comics/Blake en Mortimer/Blake En Mortimer - 01 - Het Geheim Van De Zwaardvis Deel 1.cbr&type=application/comicbook+rar"

The URL seems OK, but the file on the external share doesn't open.

Can it be the same issue?

Note: In earlier versions of Nextcloud there were no problems with external shares. It worked throughout version 27, it stopped working on version 28.

Regards, Roru69

**Comment by roru69** [comment]
For me the problem is solved by removing the "#" (asterisk) at the start of the foldername. I did this to group all external folders together but the Epubviever has a problem with decoding the # in the path.

I removed the "#" from the path and the problem was solved. Comics are opening again as usual.

Regards

RR69

**Comment by devnoname120** [comment]
@roru69 I think that the reason is that the part after the `#` is ignored because it marks the start of the URL hash tags, not the query params anymore. It would need to be escaped to avoid that issue.

**Comment by devnoname120** [comment, marked as answer]
@roru69 Your issue should be fixed in https://github.com/devnoname120/epubviewer/releases/tag/1.6.2

**Comment by roru69** [comment]
> @roru69 Your issue should be fixed in https://github.com/devnoname120/epubviewer/releases/tag/1.6.2

Correct, thanks. Everythings works as expected now.

Regards,

Roru69
